bittensor/_axon/__init__.py

```diff
@@ -339,39 +339,40 @@ def check_forward_callback( forward_callback:Callable, synapses:list = []):
         forward_callback([sample_input], synapses, hotkey='')
 
 class AuthInterceptor(grpc.ServerInterceptor):
-    """ Creates a new server interceptor that authenticates incoming messages from passed arguments.
-    """
-    def __init__(self, key:str = 'Bittensor',blacklist:List = []):
-        r""" Creates a new server interceptor that authenticates incoming messages from passed arguments.
+    """Creates a new server interceptor that authenticates incoming messages from passed arguments."""
+
+    def __init__(self, key: str = "Bittensor", blacklist: List = []):
+        r"""Creates a new server interceptor that authenticates incoming messages from passed arguments.
         Args:
             key (str, `optional`):
-                 key for authentication header in the metadata (default= Bittensor)
-            black_list (Fucntion, `optional`): 
+                 key for authentication header in the metadata (default = Bittensor)
+            black_list (Function, `optional`):
                 black list function that prevents certain pubkeys from sending messages
         """
         super().__init__()
-        self._valid_metadata = ('rpc-auth-header', key)
+        self._signature_separator = "bitxx"
```

Comment:
For the record I am not sure that separating the signature using `bitxx` has been the best design decision.

The signature contains the SS58 public key, and SS58 is based on [base58](https://datatracker.ietf.org/doc/html/draft-msporny-base58-01). The base58 alphabet contains all characters of `bitxx`, so if someone is (un)lucky enough have a public key that contains `bitxx`, the split will be wrong.

I recommend considering some other character, like `.`, as a separator,.

Reply:
hmm interesting, yea I agree that the design is flawed for unlucky individuals. I believe the math works out so that the probability of it occurring is ~1e-9. However, we have to be careful about changing it, since it can destroy all backwards compatibility.